src/components/side_nav/side_nav.tsx

```diff
@@ -150,19 +167,44 @@ export class EuiSideNav<T> extends Component<EuiSideNavProps<T>> {
 
     const nav = this.renderTree(items);
 
+    let headingNode;
+    if (heading) {
+      headingNode = (
+        <EuiTitle size="xs" className="euiSideNav__heading">
+          <h2>{heading}</h2>
+        </EuiTitle>
+      );
+
+      if (hideHeading) {
+        headingNode = <EuiScreenReaderOnly>{headingNode}</EuiScreenReaderOnly>;
+      }
+    }
+
+    const sideNavContentId = htmlIdGenerator('euiSideNavContent')();
+
     return (
       <nav className={classes} {...rest}>
         {/* Hidden from view, except in mobile */}
         <EuiButtonEmpty
           className="euiSideNav__mobileToggle"
+          textProps={{ className: 'euiSideNav__mobileToggleText' }}
+          contentProps={{ className: 'euiSideNav__mobileToggleContent' }}
           onClick={toggleOpenOnMobile}
           iconType="apps"
-          iconSide="right">
-          {mobileTitle}
+          iconSide="right"
+          aria-controls={sideNavContentId}
+          aria-expanded={isOpenOnMobile}
+          aria-haspopup="true">
+          {/* Inline h2 ensures truncation */}
+          {mobileTitle || <h2 className="eui-displayInline">{heading}</h2>}
```

Comment:
```suggestion
          {mobileTitle || <h2 id={uniqueId} className="eui-displayInline">{heading}</h2>}
```

Doing this with the `mobileTitle` is tricky but we can at least do it with the `h2` that we control: setting up a `uniqueId` and adding an `aria-labelledby={uniqueId}` to the wrapping `nav` element will ensure that the `nav` is always properly named regardless of browser/assistive tech.